Examples/Algorithms/TrackFindingExaTrkX/src/TruthGraphBuilder.cpp

```diff
@@ -0,0 +1,195 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2022 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include <Acts/Definitions/Units.hpp>
+#include <ActsExamples/TrackFindingExaTrkX/TruthGraphBuilder.hpp>
+
+using namespace Acts::UnitLiterals;
+
+namespace ActsExamples {
+
+ActsExamples::TruthGraphBuilder::TruthGraphBuilder(Config config,
+                                                   Acts::Logging::Level level)
+    : ActsExamples::IAlgorithm("TruthGraphBuilder", level),
+      m_cfg(std::move(config)) {
+  m_inputSpacePoints.initialize(m_cfg.inputSpacePoints);
+  m_inputParticles.initialize(m_cfg.inputParticles);
+  m_outputGraph.initialize(m_cfg.outputGraph);
+
+  m_inputMeasParticlesMap.maybeInitialize(m_cfg.inputMeasurementParticlesMap);
+  m_inputSimhits.maybeInitialize(m_cfg.inputSimHits);
+  m_inputMeasSimhitMap.maybeInitialize(m_cfg.inputMeasurementSimHitsMap);
+
+  bool a = m_inputMeasParticlesMap.isInitialized();
+  bool b =
+      m_inputSimhits.isInitialized() && m_inputMeasSimhitMap.isInitialized();
+
+  if (!(a || b)) {
+    throw std::invalid_argument("Missing inputs, cannot build truth graph");
+  }
```

Comment:
should we also issue an exception if both is present?

Reply:
Changed to XOR operation